adapters/repos/db/replication.go

```diff
@@ -374,6 +374,15 @@ func (i *Index) IncomingListFiles(ctx context.Context,
 	defer release()
 
 	sd := backup.ShardDescriptor{Name: shardName}
+
+	// prevent writing into the index during collection of metadata
+	i.shardTransferMutex.Lock()
+	defer i.shardTransferMutex.Unlock()
+
+	if err = localShard.Store().FlushMemtables(ctx); err != nil {
+		return nil, fmt.Errorf("flush memtables: %w", err)
+	}
```

Comment:
could you add comments about what issue happens without these changes as a note to our future selves?

Reply:
added in the upcoming PR